transport-classes-io_uring/src/main/java/io/netty/channel/uring/SubmissionQueue.java

```diff
@@ -251,14 +251,36 @@ private int submit(int toSubmit, int minComplete, int flags) {
             if (ret < 0) {
                 throw new RuntimeException("ioUringEnter syscall returned " + ret);
             }
-            // some submission might fail if these are done inline and failed.
-            logger.trace("Not all submissions succeeded. Only {} of {} SQEs were submitted.", ret, toSubmit);
         }
         return ret;
     }
 
     private int ioUringEnter(int toSubmit, int minComplete, int flags) {
         int f = enterFlags | flags;
+
+        if (IoUring.isIOUringSetupSubmitAllSupported()) {
+            return ioUringEnter0(toSubmit, minComplete, f);
+        }
+        // If IORING_SETUP_SUBMIT_ALL is not supported we need to loop until we submitted everything as
+        // io_uring_enter(...) will stop submitting once the first inline executed submission fails.
+        int submitted = 0;
+        for (;;) {
+            int ret = ioUringEnter0(toSubmit, minComplete, f);
+            if (ret < 0) {
+                return ret;
+            }
+            submitted += ret;
+            if (ret == toSubmit) {
+                return submitted;
+            }
+            // some submission might fail if these are done inline and failed.
+            logger.trace("Not all submissions succeeded. Only {} of {} SQEs were submitted.", ret, toSubmit);
```

Comment:
we trust it having this in the hot path?

Reply:
Maybe we should check the logging level at least

Reply:
+100